src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -80,70 +89,184 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
+        except Exception as error:
+            self.composite_logger.log_error("[APM] Error in modifying custom sources list. [Error={0}]".format(repr(error)))
+            return str(), str()    # defaults code to safety
 
-            command = self.cmd_prep_custom_sources_list_template.replace("<CLASSIFICATION>", base_classification if base_classification == "security" else "\"\"")
-            code, out = self.env_layer.run_command_output(command, False, False)
-            sources_content = self.env_layer.file_system.read_with_retry(self.custom_sources_list)
-            self.composite_logger.log_debug("[APM] Modified custom sources list with classification. [Code={0}][Out={1}][Content={2}]".format(str(code), str(out), str(sources_content)))   # non-zero error code to be investigated
+        # Refresh repo
+        self.refresh_repo(source_parts_dir=self.current_source_parts_dir, source_list=self.current_source_list)
 
-            if max_patch_published_date != str():
-                target = "://snapshot.ubuntu.com/ubuntu/{0}".format(self.max_patch_publish_date)
+        return self.current_source_parts_dir, self.current_source_list
 
-                if "://snapshot.ubuntu.com/ubuntu/" in sources_content:
-                    sources_content_split = sources_content.split(" ")
-                    for i in range(0, len(sources_content_split)):
-                        if "://snapshot.ubuntu.com/ubuntu/" in sources_content_split[i]:
-                            sources_content.replace(sources_content_split[i], target)
+    def __get_consolidated_source_parts_content(self, max_patch_published_date, base_classification):
+        # type: (str, str) -> (str, str)
+        """ Consolidates all list and sources files into a consistent format single source list """
+        # read basic sources list - exception being thrown are acceptable
+        source_parts_list_content = str()
+        source_parts_deb882_style_content = str()
 
-                sources_content = sources_content.replace("://azure.archive.ubuntu.com/ubuntu/", target)
-                sources_content = sources_content.replace("://in.archive.ubuntu.com/ubuntu/", target)
-                sources_content = sources_content.replace("://security.ubuntu.com/ubuntu/", target)
-                sources_content = sources_content.replace("http://snapshot.ubuntu.com/", "https://snapshot.ubuntu.com/")
-                self.composite_logger.log_debug("[APM] Modified custom sources list with snapshot. [Code={0}][Out={1}][Content={2}]".format(str(code), str(out), str(sources_content)))
+        if os.path.isdir(self.APT_SOURCES_LIST_DIR_PATH):
+            # process files in directory
+            dir_path = self.APT_SOURCES_LIST_DIR_PATH
+            for file_name in os.listdir(dir_path):
+                try:
+                    file_path = os.path.join(dir_path, file_name)
+                    if os.path.isdir(file_path):
+                        continue
 
```

Comment:
Marked as uncovered by tests

Reply:
Defensive code - resolved by inverting 'if' in code for the specific 'continue' statement which will no longer exist

Reply:
**[Candidate for resolving]** Resolved in a different way.